core/tests/protractor_desktop/topicsAndSkillsDashboard.js

```diff
@@ -57,10 +54,42 @@ describe('Topics and skills dashboard functionality', function() {
 
   it('should add a new topic to list', function() {
     topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
-    topicsAndSkillsDashboardPage.createTopic('Topic 1', true);
+    topicsAndSkillsDashboardPage.createTopic(
+      'Topic 1', 'Topic 1 description', 'Mathematics', true);
 
     topicsAndSkillsDashboardPage.get();
     topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(1);
+    topicsAndSkillsDashboardPage.deleteTopicWithIndex(0);
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+  });
+
+  it('should filter the topics', function() {
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    topicsAndSkillsDashboardPage.createTopic(
+      'Alpha', 'Alpha description', 'Mathematics', true);
+    topicsAndSkillsDashboardPage.get();
+    topicsAndSkillsDashboardPage.createTopic(
+      'Beta', 'Beta description', 'Mathematics', true);
+
+    topicsAndSkillsDashboardPage.get();
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(2);
+
+    topicsAndSkillsDashboardPage.filterTopicsByKeyword('alp');
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(1);
+    topicsAndSkillsDashboardPage.resetTopicFilters();
+
+    topicsAndSkillsDashboardPage.filterTopicsByKeyword('be');
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(1);
+    topicsAndSkillsDashboardPage.resetTopicFilters();
+
+    topicsAndSkillsDashboardPage.filterTopicsByKeyword('gamma');
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    topicsAndSkillsDashboardPage.resetTopicFilters();
+
+    topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(2);
+
+    topicsAndSkillsDashboardPage.filterTopicsByCategory('Mathematics');
```

Comment:
Each action should cause the "before" and "after" assertion to change, otherwise it is not clear that anything happened or that the functionality is working correctly. Please set the test up so that there is a meaningful difference between the before/after assertions.

Reply:
Updated.